controllers/restic.go

```diff
@@ -134,125 +139,97 @@ func (r *VeleroReconciler) buildResticDaemonset(velero *oadpv1alpha1.Velero, ds
 	if ds == nil {
 		return nil, fmt.Errorf("ds cannot be nil")
 	}
-	ds.Spec = appsv1.DaemonSetSpec{
-		Selector: ds.Spec.Selector,
-		UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
-			Type: appsv1.RollingUpdateDaemonSetStrategyType,
-		},
-		Template: v1.PodTemplateSpec{
-			ObjectMeta: metav1.ObjectMeta{
-				Labels: map[string]string{
-					"component": Restic,
-				},
+
+	resticDaemonSetName := ds.Name
+
+	*ds = *install.DaemonSet(ds.Namespace,
```

Comment:
Had same concern on Tiger's PR this would be overriding stuff we didn't want it to but 99% sure this is safe since we are constructing the DS the same each time